pkg/fission-cli/spec.go

```diff
@@ -496,6 +497,11 @@ func (fr *FissionResources) parseYaml(b []byte, loc *location) error {
 		if err != nil {
 			return errors.Wrap(err, fmt.Sprintf("Failed to parse %v in %v", tm.Kind, loc))
 		}
+		regex := regexp.MustCompile("^[a-zA-Z0-9_.]*$")
+		if !regex.MatchString(v.Spec.Package.FunctionName) {
```

Comment:
Since we are not able to 100% guarantee that no existing spec works after adding this check.
It's better to put the check at `https://github.com/fission/fission/blob/88b534377540995cbaaac0175ad0845378456957/pkg/fission-cli/spec.go#L438` and shows a warning message instead of returning an error here.
It would be great to add a TODO mark to remind us to abort requests after couple releases.
```
// TODO: xxxxxx
```

Reply:
Sounds good and changes are done.